### PR TITLE
feat(test): Fix console.error getting swallowed up

### PR DESCRIPTION
**Pull request by billyvg**
Oops I seemed to have broken this awhile back so we have been swallowing `console.error()` calls that pass in `Error` objects (i.e. React warnings call it with strings). I've removed the custom code and added https://github.com/ValentinH/jest-fail-on-console

Thanks to @eliashussary for discovering this.

Requires https://github.com/getsentry/getsentry/pull/8303

### CODE REVIEW

File: static/app/components/modals/widgetViewerModal.spec.tsx
Comment (billyvg) [inline static/app/components/modals/widgetViewerModal.spec.tsx:1039]: debugged a bit and it seemed like state was stabilized before unmounting -- gave up...

File: static/app/locale.tsx
Comment (billyvg) [inline static/app/locale.tsx:75]: I tried setting locale in test setup files, but it was still erroring in certain tests. Removing because it was kind of useless anyway (and we're capturing in Sentry).

File: static/app/utils/useMetricsContext.spec.tsx
Comment (billyvg) [inline static/app/utils/useMetricsContext.spec.tsx]: These types of tests will fail so we have to mock `console.error`

File: static/app/views/dashboardsV2/dashboard.spec.tsx
Comment (billyvg) [inline static/app/views/dashboardsV2/dashboard.spec.tsx:301]: heads up @getsentry/discover-n-dashboards if you see `act` errors, this is how I fixed them

File: static/app/views/dashboardsV2/detail.spec.jsx
Comment (billyvg) [inline static/app/views/dashboardsV2/detail.spec.jsx:25]: This was complaining about setState after unmounting... none of the tests seemed to need the actual component so...

File: static/app/views/dashboardsV2/widgetBuilder/widgetBuilderDataset.spec.tsx
Comment (billyvg) [inline static/app/views/dashboardsV2/widgetBuilder/widgetBuilderDataset.spec.tsx:27]: It was complaining that `WorldMapChart` was not returning anything.

File: static/app/views/eventsV2/queryList.spec.jsx
Comment (billyvg) [inline static/app/views/eventsV2/queryList.spec.jsx:14]: One of the deeper components that used `eventsRequest` was erroring because a response was not a list (it was trying to access `.length`). Mocking since none of the tests seem to depend on it.

File: static/app/views/sharedGroupDetails/index.spec.tsx
Comment (billyvg) [inline static/app/views/sharedGroupDetails/index.spec.tsx:31]: There's a component down in this tree that uses `useRouteContext`.

**Review by billyvg — COMMENTED** [review]
You may want to hide whitespace when reviewing:
image

File: static/app/locale.tsx
Comment (evanpurkhiser) [reply static/app/locale.tsx:75]: oh my god 🙏

**Comment by billyvg** [comment]
Not sure why self-hosted is failing, will look at it after the getsentry PR is merged.

File: static/app/views/dashboardsV2/widgetCard/widgetCard.spec.tsx
Comment (billyvg) [reply static/app/views/dashboardsV2/widgetCard/widgetCard.spec.tsx:462]: iirc state wasn't fully stabilized after render. Most of these places were complaining `act` and most places required a tick because state was being updated during mount.

**Review by eliashussary — APPROVED** [review]
huge improvement for our dev sanity :)

File: config/tsconfig.build.json
Comment (billyvg) [inline config/tsconfig.build.json:11]: So `tsconfig.build.json` is now only called when building assets from `setup.py` -- intent is to have it only type application-code and not test-code. Now that tests are co-located, test utility files do not have a pattern right now that we can catch and exclude.

any ideas here? cc @scttcper @evanpurkhiser

Comment (billyvg) [reply config/tsconfig.build.json:11]: I'm gonna go ahead and merge this, I can follow-up with this if needed

**Pull request merged**
